test/Microsoft.AspNetCore.HttpsEnforcement.Tests/HttpsPolicyMiddlewareTests.cs

```diff
@@ -0,0 +1,257 @@
+                })
+                .Configure(app =>
+                {
+                    app.UseHttpsPolicy();
+                    app.UseHsts();
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello world");
+                    });
+                });
+
+            var server = new TestServer(builder);
+            var client = server.CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            Assert.Equal(statusCode, (int)response.StatusCode);
+            Assert.Equal(expectedUrl, response.Headers.Location.ToString());
+
+            client = server.CreateClient();
+            client.BaseAddress = new Uri(response.Headers.Location.ToString());
+            request = new HttpRequestMessage(HttpMethod.Get, "");
+            response = await client.SendAsync(request);
+
+            Assert.Equal(expectedHstsHeader, response.Headers.GetValues(HeaderNames.StrictTransportSecurity).FirstOrDefault());
+        }
+
+        [Theory]
+        [InlineData(302, null, 0, false, false, "max-age=0", "https://localhost/")]
+        [InlineData(301, 5050, 0, false, false, "max-age=0", "https://localhost:5050/")]
+        [InlineData(301, 443, 0, false, false, "max-age=0", "https://localhost/")]
+        [InlineData(301, 443, 0, true, false, "max-age=0; includeSubDomains", "https://localhost/")]
+        [InlineData(301, 443, 0, false, true, "max-age=0; preload", "https://localhost/")]
+        [InlineData(301, null, 0, true, true, "max-age=0; includeSubDomains; preload", "https://localhost/")]
+        [InlineData(302, 5050, 0, true, true, "max-age=0; includeSubDomains; preload", "https://localhost:5050/")]
+        public async Task SetOptions_AddHstsMiddlewareAndEnableInPolicy(int statusCode, int? tlsPort, int maxAge, bool includeSubDomains, bool preload, string expectedHstsHeader, string expectedUrl)
+        {
+
+            // Note in this scenario the middleware is added twice.
+            var builder = new WebHostBuilder()
+                .UseUrls("https://*:5050", "http://*:5050")
+                .ConfigureServices(services =>
+                {
+                    services.Configure<HttpsPolicyOptions>(options =>
+                    {
+                        options.SetHsts = true;
+                        options.StatusCode = statusCode;
+                        options.TlsPort = tlsPort;
+                    });
+                    services.Configure<HstsOptions>(options =>
+                    {
+                        options.IncludeSubDomains = includeSubDomains;
+                        options.MaxAge = maxAge;
+                        options.Preload = preload;
+                    });
+                })
+                .Configure(app =>
+                {
+                    app.UseHttpsPolicy();
+                    app.UseHsts();
```

Comment:
I thought UseHttpsPolicy would add Hsts if needed, why is this line here. Is that not the case?

Reply:
That is the case. This is just showing that you can still add Hsts again after calling UseHttps.